readthedocs/projects/models.py

```diff
@@ -804,7 +833,10 @@ def vcs_repo(self, version=LATEST, environment=None):
         if not backend:
             repo = None
         else:
-            repo = backend(self, version, environment)
+            repo = backend(
+                self, version, environment=environment,
+                verbose_name=verbose_name, version_type=version_type
```

Comment:
I feel that we are passing `version_type` all around the code. Although, I don't have a better pattern in mind now :cry:

Reply:
yeah :(